app/controllers/admin/persons_controller.rb

```diff
@@ -49,6 +49,66 @@ def results
       )
     end
 
+    def registrations
+      person_wca_id = params.require(:wcaId)
+
+      person = Person.current.find_by!(wca_id: person_wca_id)
+      registrations = person.user.registrations
+                            .joins(:competition)
+                            .where('competitions.start_date > ?', Date.current)
+                            .order('competitions.start_date DESC')
+                            .select(
+                              'registrations.*,
+          competitions.id   AS competition_id,
+          competitions.name AS competition_name,
+          competitions.city_name,
+          competitions.country_id,
+          competitions.start_date',
+                            )
+
+      render json: registrations.as_json(
+        only: %w[competition_id competition_name city_name country_id start_date competing_status],
+      )
+    end
+
+    def organized_competitions
+      person_wca_id = params.require(:wcaId)
+
+      person = Person.current.find_by!(wca_id: person_wca_id)
+      competitions = person.user.organized_competitions
+                           .order('competitions.start_date DESC')
+                           .where.not(competitions: { announced_at: nil })
+                           .where(competitions: { cancelled_at: nil })
+                           .select(
+                             'competitions.*,
+          competitions.id   AS competition_id,
+          competitions.name AS competition_name',
+                           )
+
+      render json: competitions.as_json(
+        only: %w[competition_id competition_name city_name country_id start_date competing_status],
+      )
+    end
+
+    def delegated_competitions
+      person_wca_id = params.require(:wcaId)
+
+      person = Person.current.find_by!(wca_id: person_wca_id)
+      competitions = person.user.delegated_competitions
+                           .order('competitions.start_date DESC')
+                           .where.not(competitions: { announced_at: nil })
+                           .where(competitions: { cancelled_at: nil })
+                           .select(
+                             'competitions.*,
+          competitions.id   AS competition_id,
+          competitions.name AS competition_name',
+                           )
```

Comment:
```suggestion
                           .over.visible.not_cancelled
                           .order(start_date: :desc)
```

Reply:
I don't think your renaming is necessary if you're only loading the competition models and nothing else. You can just use the existing `id` and `name` fields in that case.